ESP/tools/createzip.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+Import("env")
+
+import sys
+from ntpath import basename
+from zipfile import ZipFile
+import json
+
+
+def createZip(source, target, env):
+    if (
+        sys.platform.startswith("ubuntu")
+        or sys.platform.startswith("Linux")
+        or sys.platform.startswith("linux")
+    ):
+        print("Program has been built, creating zip archive!")
+        program_path = target[0].get_abspath()
+        #print("binary path", program_path)
+        #print("FLASH_EXTRA_IMAGES: %s\n" % env["FLASH_EXTRA_IMAGES"])
+        #print("ESP32_APP_OFFSET: %s\n" % env["ESP32_APP_OFFSET"])
+        
+        array_args = [env["FLASH_EXTRA_IMAGES"]]
+
+        for [offset, image] in env["FLASH_EXTRA_IMAGES"]:
+            print("\nImage: %s" % str(image))
+            array_args.append(str(offset))
+            array_args.append(str(image))
+
+        array_args.append(env["ESP32_APP_OFFSET"])
+        array_args.append(program_path)
+
+        n = 2
+        partitions_arg = array_args[1:]
+        print("partitions_arg: %s\n" % partitions_arg)
```

Comment:
we could use an Fstring here, they make things a bit more readable by presenting how the output will look like exactly, for example:

```python
print(f"partitions_args: {partitions_arg}") 
```

The general rule of thumb with them goes a bit like this: If it's a pretty simple case, like printing a variable, or a function with one argument - it's safe to use an F`String, it will make things more readable, otherwise I'd stick with good ol format :D

Reply:
Haha, yeah, I forget Fstrings exist. I am too used to format specifiers from cpp.